authority/admin/api/provisioner.go

```diff
@@ -0,0 +1,175 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"github.com/smallstep/certificates/api"
+	"github.com/smallstep/certificates/authority"
+	"github.com/smallstep/certificates/authority/admin"
+	"github.com/smallstep/certificates/authority/provisioner"
+	"github.com/smallstep/certificates/errs"
+	"go.step.sm/linkedca"
+)
+
+// GetProvisionersResponse is the type for GET /admin/provisioners responses.
+type GetProvisionersResponse struct {
+	Provisioners provisioner.List `json:"provisioners"`
+	NextCursor   string           `json:"nextCursor"`
+}
+
+// GetProvisioner returns the requested provisioner, or an error.
+func (h *Handler) GetProvisioner(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	id := r.URL.Query().Get("id")
+	name := chi.URLParam(r, "name")
+
+	var (
+		p   provisioner.Interface
+		err error
+	)
+	if len(id) > 0 {
+		if p, err = h.auth.LoadProvisionerByID(id); err != nil {
+			api.WriteError(w, admin.WrapErrorISE(err, "error loading provisioner %s", id))
+			return
+		}
+	} else {
+		if p, err = h.auth.LoadProvisionerByName(name); err != nil {
+			api.WriteError(w, admin.WrapErrorISE(err, "error loading provisioner %s", name))
+			return
+		}
+	}
+
+	prov, err := h.db.GetProvisioner(ctx, p.GetID())
+	if err != nil {
+		api.WriteError(w, err)
+		return
+	}
+	api.ProtoJSON(w, prov)
+}
+
+// GetProvisioners returns the given segment of  provisioners associated with the authority.
+func (h *Handler) GetProvisioners(w http.ResponseWriter, r *http.Request) {
+	cursor, limit, err := api.ParseCursor(r)
+	if err != nil {
+		api.WriteError(w, admin.WrapError(admin.ErrorBadRequestType, err,
+			"error parsing cursor & limit query params"))
+		return
+	}
+
+	p, next, err := h.auth.GetProvisioners(cursor, limit)
+	if err != nil {
+		api.WriteError(w, errs.InternalServerErr(err))
+		return
+	}
+	api.JSON(w, &GetProvisionersResponse{
+		Provisioners: p,
+		NextCursor:   next,
+	})
```

Comment:
Provisioners won't be nicely marshaled without using protojson.

Reply:
The provisioners in this list aren't protojson linkedca provisioners anyway. They're the certificates type of provisioner. This method isn't being used right now. We're still using the existing API with `step ca provisioner list`.

I can take this method out for now if you want.

Reply:
Ok for now, but to get a nicer JSON the same that we can use [ProvisionerList](https://github.com/smallstep/linkedca/blob/27424aae8d2588aad217c75897cba5eebcb9aa91/provisioners.proto#L50-L52) with a cursor. 

It's also possible to add custom MarshalJSON and UnmarshalJSON to GetProvisionersResponse that uses protojson to marshal the list of Provisioners/Admins.